Source/WebCore/platform/SharedBuffer.cpp

```diff
@@ -67,6 +78,32 @@ Ref<FragmentedSharedBuffer> FragmentedSharedBuffer::create(DataSegment::Provider
     return adoptRef(*new FragmentedSharedBuffer(WTFMove(provider)));
 }
 
+#if PLATFORM(COCOA)
+std::optional<Ref<FragmentedSharedBuffer>> FragmentedSharedBuffer::fromIPCData(IPCData&& ipcData)
+{
+    return WTF::switchOn(WTFMove(ipcData), [](Vector<std::span<const uint8_t>>&& data) -> std::optional<Ref<FragmentedSharedBuffer>> {
+        size_t size = sizeof(data[0]) * data.size();
+        if (useUnixDomainSockets() || size < minimumPageSize) {
+            Ref sharedMemoryBuffer = SharedBuffer::create();
```

Comment:
This needs to construct a `FragmentedSharedBuffer`, not a `SharedBuffer` in the case there is more than one segment. We may want to optimize the case where there is a single vector and return `SharedBuffer::create(firstSegment)` in that case though.